src/Analysis/Core/Impl/Interpreter/PythonLibraryPath.cs

```diff
@@ -170,5 +203,91 @@ public override string ToString()
                 fs.DeleteDirectory(tempWorkingDir, true);
             }
         }
+
+        public static (IReadOnlyList<PythonLibraryPath> interpreterPaths, IReadOnlyList<PythonLibraryPath> userPaths) ClassifyPaths(
+            string root,
+            IFileSystem fs,
+            IEnumerable<PythonLibraryPath> fromInterpreter,
+            IEnumerable<string> fromUser
+        ) {
+#if DEBUG
+            Debug.Assert(root == null || root.PathEquals(PathUtils.NormalizePathAndTrim(root)));
+            Debug.Assert(!fromInterpreter.Any(p => !p.Path.PathEquals(PathUtils.NormalizePathAndTrim(p.Path))));
+#endif
+
+            // Clean up user configured paths.
+            // 1) Normalize paths.
+            // 2) If a path isn't rooted, then root it relative to the workspace root. If there is no root, just continue.
+            // 3) Trim off any ending separators for consistency.
+            // 4) Remove any empty paths, FS root paths (bad idea), or paths equal to the root.
+            // 5) Deduplicate, preserving the order specified by the user.
+            var fromUserList = fromUser
+                .Select(PathUtils.NormalizePath)
+                .Select(p => root == null || IOPath.IsPathRooted(p) ? p : IOPath.GetFullPath(IOPath.Combine(root, p))) // TODO: Replace with GetFullPath(p, root) when .NET Standard 2.1 is out.
+                .Select(PathUtils.TrimEndSeparator)
+                .Where(p => !string.IsNullOrWhiteSpace(p) && p != "/" && !p.PathEquals(root))
+                .Distinct(PathEqualityComparer.Instance)
+                .ToList();
+
+            // Remove any interpreter paths specified in the user config so they can be reclassified.
+            // The user list is usually small; List.Contains should not be too slow.
+            fromInterpreter = fromInterpreter.Where(p => !fromUserList.Contains(p.Path, PathEqualityComparer.Instance));
+
+            var stdlibLookup = fromInterpreter.ToLookup(p => p.Type == PythonLibraryPathType.StdLib);
```

Comment:
FYI: We have `EnumerableExtensions.Split`, but it's up to you.

Reply:
Ah, I figured we had something like that but couldn't find it.